controllers/component_controller.go

```diff
@@ -328,6 +329,40 @@ func (r *ComponentReconciler) runBuild(ctx context.Context, component *appstudio
 
 	log := r.Log.WithValues("Namespace", component.Namespace, "Application", component.Spec.Application, "Component", component.Name)
 
+	// Not exposing this as config on purpose since this is supposed to be moved out eventually
+	// Also, this is really being used for tests.
+	credentialPath := os.Getenv("BUILD_CREDENTIALS")
+	if credentialPath == "" {
+		// expect this to be unset in production.
+		credentialPath = "/etc/build/.dockerconfigjson"
+	}
+
+	if strings.Contains(component.Spec.Build.ContainerImage, r.ImageRepository) {
+
+		defaultSecret := gitops.DefaultUserWorkloadPullSecret(*component, credentialPath)
+		err := controllerutil.SetOwnerReference(component, &defaultSecret, r.Scheme)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("Unable to set owner reference for %v", defaultSecret))
+			return err
+		}
+		secret := &corev1.Secret{}
+		err = r.Get(ctx, types.NamespacedName{Name: defaultSecret.Name, Namespace: component.Namespace}, secret)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				err = r.Client.Create(ctx, &defaultSecret)
```

Comment:
When the component gets deleted, we need to clean the secret too.

Reply:
Yes, it should get cleaned by `ownerReferences`